README.rst
Introduction
------------

This is a `Django CMS` plugin for embedding a Google Document Form to a page by using the form's variable formkey. 

Google Forms is part of Google's Documents application suite and automatically links a dynamic spreadsheet that updates when a form is submitted. 
For more info on how to use the form application go to http://www.google.com/google-d-s/forms/

Installation
------------

First, you must install `Django CMS` which require `Django`. For full details, see the installation 
instructions for those packages.

Install ``cmsplugin-googleform`` to your environment with a tool such as `PIP`, 
`setuptools`, or `buildout`.

Add ``cmsplugin_googleform`` to the ``INSTALLED_APPS`` list in your project's 
``settings.py`` and run the ``syncdb`` command on your ``manage.py``.

.. _Django: http://www.djangoproject.com/
.. _Django CMS: https://www.django-cms.org/
.. _PIP: http://www.pip-installer.org/
.. _setuptools: http://pypi.python.org/pypi/setuptools/
.. _buildout: http://pypi.python.org/pypi/zc.buildout/

What's Inside
-------------

When you add the plugin to a page, you will required to input a **Form key**. To find the ``formkey`` for a form, at the top of the form spreadsheet in Google Documents, go to ``Form > Embed form in webpage...``.
In the embed code, copy the string after ``formkey=`` and paste it in the cmsplugin-googleform **Form key** field.

**Example embed code:**
``<iframe... formkey=dG9jLXJSYUY1a3NiYTRLYWs3S1F0xxxxxxx"...</iframe>``

It is recommended that you fill out the **Height** and **Width** fields for the form (though it is not required).



HISTORY.rst
0.1 (2012-06-05)
----------------

* First Release



cmsplugin_googleform/migrations/0001_initial.py
# encoding: utf-8
import datetime
from south.db import db
from south.v2 import SchemaMigration
from django.db import models

class Migration(SchemaMigration):

    def forwards(self, orm):
        
        # Adding model 'GoogleFormsPlugin'
        db.create_table('cmsplugin_googleformsplugin', (
            ('cmsplugin_ptr', self.gf('django.db.models.fields.related.OneToOneField')(to=orm['cms.CMSPlugin'], unique=True, primary_key=True)),
            ('form_id', self.gf('django.db.models.fields.CharField')(max_length=64)),
            ('template', self.gf('django.db.models.fields.CharField')(max_length=255, null=True, blank=True)),
        ))
        db.send_create_signal('cmsplugin_googleform', ['GoogleFormsPlugin'])


    def backwards(self, orm):
        
        # Deleting model 'GoogleFormsPlugin'
        db.delete_table('cmsplugin_googleformsplugin')


    models = {
        'cms.cmsplugin': {
            'Meta': {'object_name': 'CMSPlugin'},
            'creation_date': ('django.db.models.fields.DateTimeField', [], {'default': 'datetime.datetime.now'}),
            'id': ('django.db.models.fields.AutoField', [], {'primary_key': 'True'}),
            'language': ('django.db.models.fields.CharField', [], {'max_length': '15', 'db_index': 'True'}),
            'level': ('django.db.models.fields.PositiveIntegerField', [], {'db_index': 'True'}),
            'lft': ('django.db.models.fields.PositiveIntegerField', [], {'db_index': 'True'}),
            'parent': ('django.db.models.fields.related.ForeignKey', [], {'to': "orm['cms.CMSPlugin']", 'null': 'True', 'blank': 'True'}),
            'placeholder': ('django.db.models.fields.related.ForeignKey', [], {'to': "orm['cms.Placeholder']", 'null': 'True'}),
            'plugin_type': ('django.db.models.fields.CharField', [], {'max_length': '50', 'db_index': 'True'}),
            'position': ('django.db.models.fields.PositiveSmallIntegerField', [], {'null': 'True', 'blank': 'True'}),
            'rght': ('django.db.models.fields.PositiveIntegerField', [], {'db_index': 'True'}),
            'tree_id': ('django.db.models.fields.PositiveIntegerField', [], {'db_index': 'True'})
        },
        'cms.placeholder': {
            'Meta': {'object_name': 'Placeholder'},
            'default_width': ('django.db.models.fields.PositiveSmallIntegerField', [], {'null': 'True'}),
            'id': ('django.db.models.fields.AutoField', [], {'primary_key': 'True'}),
            'slot': ('django.db.models.fields.CharField', [], {'max_length': '50', 'db_index': 'True'})
        },
        'cmsplugin_googleform.googleformsplugin': {
            'Meta': {'object_name': 'GoogleFormsPlugin', 'db_table': "'cmsplugin_googleformsplugin'", '_ormbases': ['cms.CMSPlugin']},
            'cmsplugin_ptr': ('django.db.models.fields.related.OneToOneField', [], {'to': "orm['cms.CMSPlugin']", 'unique': 'True', 'primary_key': 'True'}),
            'form_id': ('django.db.models.fields.CharField', [], {'max_length': '64'}),
            'template': ('django.db.models.fields.CharField', [], {'max_length': '255', 'null': 'True', 'blank': 'True'})
        }
    }

    complete_apps = ['cmsplugin_googleform']


cmsplugin_googleform/migrations/0002_add_dimensions.py
# encoding: utf-8
import datetime
from south.db import db
from south.v2 import SchemaMigration
from django.db import models

class Migration(SchemaMigration):

    def forwards(self, orm):
        
        # Deleting field 'GoogleFormsPlugin.template'
        db.delete_column('cmsplugin_googleformsplugin', 'template')

        # Adding field 'GoogleFormsPlugin.height'
        db.add_column('cmsplugin_googleformsplugin', 'height', self.gf('django.db.models.fields.PositiveIntegerField')(null=True, blank=True), keep_default=False)

        # Adding field 'GoogleFormsPlugin.width'
        db.add_column('cmsplugin_googleformsplugin', 'width', self.gf('django.db.models.fields.PositiveIntegerField')(null=True, blank=True), keep_default=False)


    def backwards(self, orm):
        
        # Adding field 'GoogleFormsPlugin.template'
        db.add_column('cmsplugin_googleformsplugin', 'template', self.gf('django.db.models.fields.CharField')(max_length=255, null=True, blank=True), keep_default=False)

        # Deleting field 'GoogleFormsPlugin.height'
        db.delete_column('cmsplugin_googleformsplugin', 'height')

        # Deleting field 'GoogleFormsPlugin.width'
        db.delete_column('cmsplugin_googleformsplugin', 'width')


    models = {
        'cms.cmsplugin': {
            'Meta': {'object_name': 'CMSPlugin'},
            'creation_date': ('django.db.models.fields.DateTimeField', [], {'default': 'datetime.datetime.now'}),
            'id': ('django.db.models.fields.AutoField', [], {'primary_key': 'True'}),
            'language': ('django.db.models.fields.CharField', [], {'max_length': '15', 'db_index': 'True'}),
            'level': ('django.db.models.fields.PositiveIntegerField', [], {'db_index': 'True'}),
            'lft': ('django.db.models.fields.PositiveIntegerField', [], {'db_index': 'True'}),
            'parent': ('django.db.models.fields.related.ForeignKey', [], {'to': "orm['cms.CMSPlugin']", 'null': 'True', 'blank': 'True'}),
            'placeholder': ('django.db.models.fields.related.ForeignKey', [], {'to': "orm['cms.Placeholder']", 'null': 'True'}),
            'plugin_type': ('django.db.models.fields.CharField', [], {'max_length': '50', 'db_index': 'True'}),
            'position': ('django.db.models.fields.PositiveSmallIntegerField', [], {'null': 'True', 'blank': 'True'}),
            'rght': ('django.db.models.fields.PositiveIntegerField', [], {'db_index': 'True'}),
            'tree_id': ('django.db.models.fields.PositiveIntegerField', [], {'db_index': 'True'})
        },
        'cms.placeholder': {
            'Meta': {'object_name': 'Placeholder'},
            'default_width': ('django.db.models.fields.PositiveSmallIntegerField', [], {'null': 'True'}),
            'id': ('django.db.models.fields.AutoField', [], {'primary_key': 'True'}),
            'slot': ('django.db.models.fields.CharField', [], {'max_length': '50', 'db_index': 'True'})
        },
        'cmsplugin_googleform.googleformsplugin': {
            'Meta': {'object_name': 'GoogleFormsPlugin', 'db_table': "'cmsplugin_googleformsplugin'", '_ormbases': ['cms.CMSPlugin']},
            'cmsplugin_ptr': ('django.db.models.fields.related.OneToOneField', [], {'to': "orm['cms.CMSPlugin']", 'unique': 'True', 'primary_key': 'True'}),
            'form_id': ('django.db.models.fields.CharField', [], {'max_length': '64'}),
            'height': ('django.db.models.fields.PositiveIntegerField', [], {'null': 'True', 'blank': 'True'}),
            'width': ('django.db.models.fields.PositiveIntegerField', [], {'null': 'True', 'blank': 'True'})
        }
    }

    complete_apps = ['cmsplugin_googleform']
